From dd7f23b7c28e0f6b74ae6f7ca6f8163695086e1f Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Sat, 9 Jun 2018 16:10:06 +0300
Subject: [PATCH 3/6] power: Remove support for msm-dcvs governor

* This is not used by any recent qcom SoC
  and not used by any SoC using this HAL.

Change-Id: If9b2997104bcd5e3ff27fc39c75590d07e813f85
---
 power-common.h |   5 --
 power-helper.c | 150 -------------------------------------------------
 utils.c        |   6 --
 utils.h        |   1 -
 4 files changed, 162 deletions(-)

diff --git a/power-common.h b/power-common.h
index 601eb42..161e0bb 100644
--- a/power-common.h
+++ b/power-common.h
@@ -30,14 +30,9 @@
 #define NODE_MAX (64)
 
 #define SCALING_GOVERNOR_PATH "/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor"
-#define DCVS_CPU0_SLACK_MAX_NODE "/sys/module/msm_dcvs/cores/cpu0/slack_time_max_us"
-#define DCVS_CPU0_SLACK_MIN_NODE "/sys/module/msm_dcvs/cores/cpu0/slack_time_min_us"
-#define MPDECISION_SLACK_MAX_NODE "/sys/module/msm_mpdecision/slack_time_max_us"
-#define MPDECISION_SLACK_MIN_NODE "/sys/module/msm_mpdecision/slack_time_min_us"
 #define SCALING_MIN_FREQ "/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq"
 #define ONDEMAND_GOVERNOR "ondemand"
 #define INTERACTIVE_GOVERNOR "interactive"
-#define MSMDCVS_GOVERNOR "msm-dcvs"
 
 #define HINT_HANDLED (0)
 #define HINT_NONE (-1)
diff --git a/power-helper.c b/power-helper.c
index f037be7..483cfe7 100644
--- a/power-helper.c
+++ b/power-helper.c
@@ -146,11 +146,6 @@ struct stat_pair wlan_stat_map[] = {
 };
 #endif
 
-static int saved_dcvs_cpu0_slack_max = -1;
-static int saved_dcvs_cpu0_slack_min = -1;
-static int saved_mpdecision_slack_max = -1;
-static int saved_mpdecision_slack_min = -1;
-static int slack_node_rw_failed = 0;
 static int display_hint_sent;
 
 void power_init(void)
@@ -317,8 +312,6 @@ extern void power_set_interactive_ext(int on);
 void power_set_interactive(int on)
 {
     char governor[80];
-    char tmp_str[NODE_MAX];
-    int rc = 0;
 
     if (!on) {
         /* Send Display OFF hint to perf HAL */
@@ -365,98 +358,6 @@ void power_set_interactive(int on)
 
             perform_hint_action(DISPLAY_STATE_HINT_ID,
                     resource_values, ARRAY_SIZE(resource_values));
-        } else if (is_msmdcvs_governor(governor)) {
-            /* Display turned off. */
-            if (sysfs_read(DCVS_CPU0_SLACK_MAX_NODE, tmp_str, NODE_MAX - 1)) {
-                if (!slack_node_rw_failed) {
-                    ALOGE("Failed to read from %s", DCVS_CPU0_SLACK_MAX_NODE);
-                }
-
-                rc = 1;
-            } else {
-                saved_dcvs_cpu0_slack_max = atoi(tmp_str);
-            }
-
-            if (sysfs_read(DCVS_CPU0_SLACK_MIN_NODE, tmp_str, NODE_MAX - 1)) {
-                if (!slack_node_rw_failed) {
-                    ALOGE("Failed to read from %s", DCVS_CPU0_SLACK_MIN_NODE);
-                }
-
-                rc = 1;
-            } else {
-                saved_dcvs_cpu0_slack_min = atoi(tmp_str);
-            }
-
-            if (sysfs_read(MPDECISION_SLACK_MAX_NODE, tmp_str, NODE_MAX - 1)) {
-                if (!slack_node_rw_failed) {
-                    ALOGE("Failed to read from %s", MPDECISION_SLACK_MAX_NODE);
-                }
-
-                rc = 1;
-            } else {
-                saved_mpdecision_slack_max = atoi(tmp_str);
-            }
-
-            if (sysfs_read(MPDECISION_SLACK_MIN_NODE, tmp_str, NODE_MAX - 1)) {
-                if(!slack_node_rw_failed) {
-                    ALOGE("Failed to read from %s", MPDECISION_SLACK_MIN_NODE);
-                }
-
-                rc = 1;
-            } else {
-                saved_mpdecision_slack_min = atoi(tmp_str);
-            }
-
-            /* Write new values. */
-            if (saved_dcvs_cpu0_slack_max != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", 10 * saved_dcvs_cpu0_slack_max);
-
-                if (sysfs_write(DCVS_CPU0_SLACK_MAX_NODE, tmp_str) != 0) {
-                    if (!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", DCVS_CPU0_SLACK_MAX_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_dcvs_cpu0_slack_min != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", 10 * saved_dcvs_cpu0_slack_min);
-
-                if (sysfs_write(DCVS_CPU0_SLACK_MIN_NODE, tmp_str) != 0) {
-                    if(!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", DCVS_CPU0_SLACK_MIN_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_mpdecision_slack_max != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", 10 * saved_mpdecision_slack_max);
-
-                if (sysfs_write(MPDECISION_SLACK_MAX_NODE, tmp_str) != 0) {
-                    if(!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", MPDECISION_SLACK_MAX_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_mpdecision_slack_min != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", 10 * saved_mpdecision_slack_min);
-
-                if (sysfs_write(MPDECISION_SLACK_MIN_NODE, tmp_str) != 0) {
-                    if(!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", MPDECISION_SLACK_MIN_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            slack_node_rw_failed = rc;
         }
     } else {
         /* Display on. */
@@ -464,57 +365,6 @@ void power_set_interactive(int on)
             undo_hint_action(DISPLAY_STATE_HINT_ID);
         } else if (is_interactive_governor(governor)) {
             undo_hint_action(DISPLAY_STATE_HINT_ID);
-        } else if (is_msmdcvs_governor(governor)) {
-            /* Display turned on. Restore if possible. */
-            if (saved_dcvs_cpu0_slack_max != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", saved_dcvs_cpu0_slack_max);
-
-                if (sysfs_write(DCVS_CPU0_SLACK_MAX_NODE, tmp_str) != 0) {
-                    if (!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", DCVS_CPU0_SLACK_MAX_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_dcvs_cpu0_slack_min != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", saved_dcvs_cpu0_slack_min);
-
-                if (sysfs_write(DCVS_CPU0_SLACK_MIN_NODE, tmp_str) != 0) {
-                    if (!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", DCVS_CPU0_SLACK_MIN_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_mpdecision_slack_max != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", saved_mpdecision_slack_max);
-
-                if (sysfs_write(MPDECISION_SLACK_MAX_NODE, tmp_str) != 0) {
-                    if (!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", MPDECISION_SLACK_MAX_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            if (saved_mpdecision_slack_min != -1) {
-                snprintf(tmp_str, NODE_MAX, "%d", saved_mpdecision_slack_min);
-
-                if (sysfs_write(MPDECISION_SLACK_MIN_NODE, tmp_str) != 0) {
-                    if (!slack_node_rw_failed) {
-                        ALOGE("Failed to write to %s", MPDECISION_SLACK_MIN_NODE);
-                    }
-
-                    rc = 1;
-                }
-            }
-
-            slack_node_rw_failed = rc;
         }
     }
 }
diff --git a/utils.c b/utils.c
index 60c1261..13dbccc 100644
--- a/utils.c
+++ b/utils.c
@@ -233,12 +233,6 @@ int is_ondemand_governor(char* governor) {
    return 0;
 }
 
-int is_msmdcvs_governor(char* governor) {
-   if (strncmp(governor, MSMDCVS_GOVERNOR, (strlen(MSMDCVS_GOVERNOR)+1)) == 0)
-      return 1;
-   return 0;
-}
-
 #ifndef INTERACTION_BOOST
 void interaction(int UNUSED(duration), int UNUSED(num_args), int UNUSED(opt_list[]))
 {
diff --git a/utils.h b/utils.h
index 27eaf92..e443749 100644
--- a/utils.h
+++ b/utils.h
@@ -35,7 +35,6 @@ int get_scaling_governor(char governor[], int size);
 int get_scaling_governor_check_cores(char governor[], int size,int core_num);
 int is_interactive_governor(char*);
 int is_ondemand_governor(char*);
-int is_msmdcvs_governor(char*);
 
 void vote_ondemand_io_busy_off();
 void unvote_ondemand_io_busy_off();
-- 
2.17.1

